internal/operands/tekton-cleanup/reconcile.go

```diff
@@ -0,0 +1,175 @@
+// +kubebuilder:rbac:groups=cdi.kubevirt.io,resources=datavolumes,verbs=create
+// +kubebuilder:rbac:groups=subresources.kubevirt.io,resources=virtualmachines/start,verbs=update
+
+// RBAC rules for "execute-in-vm-task"
+// +kubebuilder:rbac:groups=kubevirt.io,resources=virtualmachineinstances;virtualmachines,verbs=get;list;watch
+// +kubebuilder:rbac:groups=subresources.kubevirt.io,resources=virtualmachines/restart;virtualmachines/start;virtualmachines/stop,verbs=update
+
+// RBAC rules for "generate-ssh-keys-task"
+// +kubebuilder:rbac:groups=core,resources=secrets,verbs=get;list;create;patch
+
+// RBAC rules for "modify-data-object-task"
+// +kubebuilder:rbac:groups=cdi.kubevirt.io,resources=datavolumes;datasources,verbs=get;create;delete
+// +kubebuilder:rbac:groups=core,resources=pods,verbs=create
+// +kubebuilder:rbac:groups=core,resources=persistentvolumeclaims,verbs=get;delete
+
+// RBAC rules for "modify-vm-template-task"
+// +kubebuilder:rbac:groups=template.openshift.io,resources=templates,verbs=get;list;patch;delete
+
+// RBAC rules for "wait-for-vmi-status-task"
+// +kubebuilder:rbac:groups=kubevirt.io,resources=virtualmachineinstances,verbs=get;list;watch
+
+const (
+	operandName          = "tekton-cleanup"
+	operandPipelinesName = "tekton-pipelines"
+	operandTasksName     = "tekton-tasks"
+
+	tektonCrd = "tasks.tekton.dev"
+
+	tektonDeprecated = "tekton.dev/deprecated"
+)
+
+func init() {
+	utilruntime.Must(pipeline.AddToScheme(common.Scheme))
+}
+
+func WatchClusterTypes() []operands.WatchType {
+	return []operands.WatchType{
+		{Object: &v1.ConfigMap{}},
+		{Object: &rbac.RoleBinding{}},
+		{Object: &v1.ServiceAccount{}},
+		{Object: &rbac.ClusterRole{}},
+	}
+}
+
+type tektonCleanup struct {
+}
+
+var _ operands.Operand = &tektonCleanup{}
+
+func New() operands.Operand {
+	return &tektonCleanup{}
+}
+
+func (t *tektonCleanup) Name() string {
+	return operandName
+}
+
+func (t *tektonCleanup) WatchClusterTypes() []operands.WatchType {
+	return WatchClusterTypes()
+}
+
+func (t *tektonCleanup) WatchTypes() []operands.WatchType {
+	return nil
+}
+
+func (t *tektonCleanup) Reconcile(request *common.Request) ([]common.ReconcileResult, error) {
+	err := cleanupResources(request)
+	return nil, err
+}
+
+func (t *tektonCleanup) Cleanup(request *common.Request) ([]common.CleanupResult, error) {
+	err := cleanupResources(request)
+	return nil, err
+}
+
+func cleanupResources(request *common.Request) error {
+	cleanupFuncs := []func(*common.Request) error{
+		cleanupResource[rbac.ClusterRoleList, rbac.ClusterRole],
+		cleanupResource[rbac.RoleBindingList, rbac.RoleBinding],
+		cleanupResource[v1.ServiceAccountList, v1.ServiceAccount],
+		cleanupResource[v1.ConfigMapList, v1.ConfigMap],
+	}
+	if request.CrdList.CrdExists(tektonCrd) {
+		cleanupFuncs = append(cleanupFuncs, cleanupResource[pipeline.PipelineList, pipeline.Pipeline]) //nolint:staticcheck
```

Comment:
What does the linter complain about here?

Reply:
`pipeline.Pipeline` is deprecated.

Reply:
ACK